File: backend/src/main/java/com/yourcompany/onlineexam/model/QuestionBank.java
package com.yourcompany.onlineexam.model;

public class QuestionBank {
    private String id;
    private String name;
    private String courseId;
    private String courseName;
    private String description;
    private int totalQuestions;
    private int easyCount;
    private int mediumCount;
    private int hardCount;

    public QuestionBank() {}

    public String getId() { return id; }
    public void setId(String id) { this.id = id; }

    public String getName() { return name; }
    public void setName(String name) { this.name = name; }

    public String getCourseId() { return courseId; }
    public void setCourseId(String courseId) { this.courseId = courseId; }

    public String getCourseName() { return courseName; }
    public void setCourseName(String courseName) { this.courseName = courseName; }

    public String getDescription() { return description; }
    public void setDescription(String description) { this.description = description; }

    public int getTotalQuestions() { return totalQuestions; }
    public void setTotalQuestions(int totalQuestions) { this.totalQuestions = totalQuestions; }

    public int getEasyCount() { return easyCount; }
    public void setEasyCount(int easyCount) { this.easyCount = easyCount; }

    public int getMediumCount() { return mediumCount; }
    public void setMediumCount(int mediumCount) { this.mediumCount = mediumCount; }

    public int getHardCount() { return hardCount; }
    public void setHardCount(int hardCount) { this.hardCount = hardCount; }
} 

File: backend/OnlineExam/Models/QuestionBank.cs
using Google.Cloud.Firestore;

namespace OnlineExam.Models;

[FirestoreData]
public class QuestionBank
{
    [FirestoreDocumentId]
    public string? Id { get; set; }
    [FirestoreProperty("name")]
    public string? Name { get; set; }
    [FirestoreProperty("courseId")]
    public string? CourseId { get; set; }
    [FirestoreProperty("courseName")]
    public string? CourseName { get; set; }
    [FirestoreProperty("description")]
    public string? Description { get; set; }
    [FirestoreProperty("totalQuestions")]
    public int TotalQuestions { get; set; }
    [FirestoreProperty("easyCount")]
    public int EasyCount { get; set; }
    [FirestoreProperty("mediumCount")]
    public int MediumCount { get; set; }
    [FirestoreProperty("hardCount")]
    public int HardCount { get; set; }
}
